apisix/plugins/opa/helper.lua

```diff
@@ -45,16 +52,136 @@ local function build_http_request(conf, ctx)
 end
 
 
-function _M.build_opa_input(conf, ctx, subsystem)
-    local request = build_http_request(conf, ctx)
+local function _build_http_route(conf, ctx)
+    local route_id = ctx.route_id
+    local routes = get_routes()
+
+    for _, route in ipairs(routes) do
+        if route.value.id == route_id then
+            return core.table.deepcopy(route).value
+        end
+    end
+
+    return nil
+end
+
+
+local function build_http_route(conf, ctx, remove_upstream)
+    local route = _build_http_route(conf, ctx)
+
+    if remove_upstream and route and route.upstream then
+        route.upstream = nil
+    end
+
+    return route
+end
+
+
+local function _build_http_upstream(conf, ctx)
```

Comment:
As the upstream can be changed by traffic-split, we can omit the upstream feature now?

Reply:
I have removed some of the functionality of upstream for now.